src/common/com/intellij/plugins/haxe/ide/module/HaxeModuleBuilder.java

```diff
@@ -51,6 +74,75 @@ public void moduleCreated(@NotNull Module module) {
     final CompilerModuleExtension model = (CompilerModuleExtension)CompilerModuleExtension.getInstance(module).getModifiableModel(true);
     model.setCompilerOutputPath(model.getCompilerOutputUrl());
     model.inheritCompilerOutputPath(false);
+
+    final Project project = module.getProject();
+    List<Pair<String, String>> sourcePaths = getSourcePaths();
+    String srcPath = null;
+
+    if ((sourcePaths.size() > 0)) {
+      srcPath = sourcePaths.get(0).getFirst();
+      VirtualFile dir = VirtualFileManager.getInstance().findFileByUrl(VfsUtilCore.pathToUrl(srcPath));
+
+      if (dir != null) {
+        try {
+          VirtualFile mainClassFile = dir.findFileByRelativePath("Main.hx");
+          if (mainClassFile == null) {
+            HaxeFileTemplateUtil
+              .createClass("Main", "", PsiManager.getInstance(project).findDirectory(dir), "HaxeMainClass",
+                           null);
+          }
+        }
+        catch (Exception e) {
+          e.printStackTrace();
+        }
+      }
+
+      HaxeModuleSettings moduleSettings = HaxeModuleSettings.getInstance(module);
+      moduleSettings.setMainClass("Main");
+      moduleSettings.setArguments("");
+      moduleSettings.setNmeFlags("");
+      moduleSettings.setHaxeTarget(HaxeTarget.NEKO);
```

Comment:
That is interesting, I just don't use multiple modules, so I don't know how it is in practical sense